### ISSUE TITLE
onLoad android and onBuffer IOS

### ISSUE THREAD

**Issue by Ibad9** [TheWidlarzGroup/react-native-video#3793]
### Version

6.0.0

### What platforms are you having the problem on?

iOS, Android

### Architecture

Old architecture

### What happened?

I am currently facing two issues.

1) On android if i use preroll ads the onLoad function is not called automatically and if i pause and resume the video then it gets called.
2) On ios first video is played for 2 3 seconds then ad gets played and onBuffer gets called only at the start after that it never gets called even if I seek and video is buffering.

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are: 

<Video
                            //Skippable
                            adTagUrl='https://pubads.g.doubleclick.net/gampad/ads?iu=/21775744923/external/single_preroll_skippable&sz=640x480&ciu_szs=300x250%2C728x90&gdfp_req=1&output=vast&unviewed_position_start=1&env=vp&impl=s&correlator='
                            ref={videoRef}
                            source={{ uri: 'https://sample-videos.com/video321/mp4/480/big_buck_bunny_480p_30mb.mp4' }}
                            style={styles.video}
                            resizeMode='contain'
                            onLoadStart={onLoadStart}
                            onLoad={(val)=> console.log('onLoad', val)}
                            onProgress={onProgress}
                            onBuffer={(val)=> console.log('buffer', val)}
                            onReadyForDisplay={onReadyForDisplay}
                            onReceiveAdEvent={onReceiveAdEvent}
                            muted={muted}
                            playInBackground={false}
                            playWhenInactive={false}
                            controls={false}
                            pictureInPicture={false}
                            rate={playbackRate}
                        />

**Comment by andybui2025** [comment]
The same, I have downgrade version 6.0.0-rc.0, it is work

**Comment by Meet-Kapadiya** [comment]
https://github.com/TheWidlarzGroup/react-native-video/issues/3744#issuecomment-2104520969

**Comment by freeboub** [comment]
Can you please split trhe ticket in 2 differents tickets. It will be hard to follow here.
Thank you
